server.py

```diff
@@ -0,0 +1,108 @@
+from sys import argv
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from http.cookies import SimpleCookie
+from cgi import FieldStorage
+
+from jinja2 import Environment, FileSystemLoader
+
+
+HOST = '127.0.0.1'
+PORT = 8001
+
+if len(argv) > 1:
+    PORT = int(argv[1])
+
+
+class Handler(BaseHTTPRequestHandler):
+
+    def _get_cookies(self):
+        cookies = SimpleCookie(self.headers.get('Cookie'))
+
+        if len(cookies) > 0 and int(cookies['auth'].value) > 0:
+            return True
+
+        return False
+
+    def _get_template(self, page):
+        file_loader = FileSystemLoader('templates')
+        env = Environment(loader=file_loader)
+        return env.get_template(page)
+
+    def _set_cookies(self, value):
+        cookies = SimpleCookie()
+        cookies['auth'] = value
+        self.send_header('Set-Cookie', cookies.output(header='', sep=''))
+
+    def _charge(self):
+        pass
+
+    def _error(self):
+        pass
+
+    def _form(self):
+        pass
+
+    def _log_in(self):
+        self._set_cookies(100)
+
+    def _log_out(self):
+        self._set_cookies(0)
+
+    PAGES = {
+        'charge.html': _charge,
+        'error.html': _error,
+        'form.html': _form,
+        'logIn.html': _log_in,
+        'logOut.html': _log_out,
+    }
+
+    def _set_headers(self, page=None, response_code=200, content_type='text/html'):
+        self.send_response(response_code)
+        self.send_header('Content-type', content_type)
+
+        if page is not None:
+            self.PAGES[page](self)
+
+        self.end_headers()
+
+    def do_GET(self):
+        page = self.path[1:] + '.html'
+
+        if page not in self.PAGES.keys():
+            page = 'error.html'
```

Comment:
It's nice to have a fallback error page, but you also need to set the 404 response status for that page.

Reply:
I don't exactly understand the reason of adding 404 response handler. In my case when python sees page which don't exist it just changes page on custom error page, that in future can be customized with explanation for user what's gone wrong and how to fix it. Is that some kind of convention?

Reply:
Yes, a "not found"  fallback page usually goes together with the 404 response status. It's important.

Reply:
Any further actions here?

Reply:
fixed

Reply:
`curl -X GET http://127.0.0.1:8001/favicon.ico` - still error, expected 404
`curl -X GET http://127.0.0.1:8001/something` - still 200, expected 404

Reply:
Added 404 in error's pages output